custom/icds_reports/sqldata/exports/growth.py

```diff
@@ -0,0 +1,179 @@
+from corehq.apps.locations.models import SQLLocation
+from custom.icds_reports.models import ChildHealthMonthlyView
+from custom.icds_reports.sqldata.base import IcdsSqlData
+from custom.icds_reports.utils.mixins import ExportableMixin
+from custom.icds_reports.utils import get_status, calculate_date_for_age, \
+    current_month_stunting_column, \
+    current_month_wasting_column, phone_number_function, india_now
+from dateutil.relativedelta import relativedelta
+
+
+class GrowthExport(ExportableMixin, IcdsSqlData):
+    title = 'Child Growth Tracking'
+    table_name = 'child_health_monthly_view'
+
+    def get_excel_data(self, location):
+
+        def _check_case_presence(case_id, column, data_dict):
+            return data_dict[case_id][column] if case_id in data_dict.keys() else "N/A"
+
+        filters = {
+            "month": self.config['month'],
+            "age_in_months__lte": 60,
+            "valid_in_month": 1
+        }
+        initial_month = self.config['month']
+
+        if self.loc_level == 5:
+            filters['awc_id'] = location
+            order_by = ('awc_name',)
+        elif self.loc_level == 4:
+            filters['supervisor_id'] = location
+            order_by = ('awc_name',)
+        elif self.loc_level == 3:
+            filters['block_id'] = location
+            order_by = ('supervisor_name', 'awc_name')
+        elif self.loc_level == 2:
+            filters['district_id'] = location
+            order_by = ('block_name', 'supervisor_name', 'awc_name')
+        elif self.loc_level == 1:
+            filters['state_id'] = location
+            order_by = ('district_name', 'block_name', 'supervisor_name', 'awc_name')
+        else:
+            order_by = ('state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name')
+
+        query_set_1 = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
```

Comment:
these three calls to the database table is causing code repetition can be put this in a function and call is thrice instead?

Reply:
updated!!